### DISCUSSION TITLE
DDK Filter Definitions

### DISCUSSION THREAD

**Discussion by akvas** [groops-devs/groops#239] [Ideas]
Hi all,

following a discussion (https://github.com/akvas/grates/issues/4) on DDK-filtered spherical harmonics from ICGEM, it seems that there is some ambiguity when it comes to the filter parameters. DDK1 to DDK3 are explicitly given in Kusche et al. 2009 (https://doi.org/10.1007/s00190-009-0308-3), which is also the basis of the GROOPS implementation of the filter (https://github.com/groops-devs/groops/blob/e96877132ebb5380845222e635499cc400020522/source/classes/sphericalHarmonicsFilter/sphericalHarmonicsFilterDdk.h#L60C3-L60C50). 

ICGEM, however, uses parameters which differ from this generalised power law for DDK4 and higher, which are listed in https://github.com/strawpants/GRACE-filter/tree/master/data/DDK.

Since there is no official publication for DDK4 and higher, GROOPS I suggest that GROOPS should offer consistency with ICGEM, e.g., with a new appropriately named filter class in addition to the current flexible implementation.

Please let me know what you think.

Cheers,
Andreas

**Comment by tmayerguerr** [comment]
Hi Andreas,

I would not introduce a new class, but simply make a bugfix, as the exakt definition of the DDK level is not documented in GROOPS.
```
    const std::array<Double,9> factors = {0, 1e14, 1e13, 1e12, 5e11, 1e11, 5e10, 1e10, 5e9};
    const Double factor = factors.at(level);
```
What do you think?

regards
Torsten

**Reply by tmayerguerr** [reply]
The bugfix is now in the main branch.
Torsten

**Comment by akvas** [comment]
Perfect, thank you!

Cheers,
Andreas
